### ISSUE TITLE
Put a copy of magic_enum into bundled.

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#17664]
#17491 adds support for an external `magic_enum` package. My preference would be to ingest that package into `bundled/` like for boost, and fall back to that if no external one can be found. I think that the package is useful enough that we should just *always* be able to use it.

@tamiko @luca-heltai?

**Comment by tamiko** [comment]
@bangerth Good idea but I don't think in time for the release. I'll update the Ubuntu/Debian package to install libmagicenum-dev - so it will be avaiable there as an external dependency.

**Comment by luca-heltai** [comment]
Yes, I agree. Let's bundle this for 9.8.
